### PR TITLE
Add intensity to precursor table

### PR DESCRIPTION
**Pull request by GeorgWa**
For a long time, precursor and peptide information was missing from the precursor output table.
This breaking change adds in the LFQ intensities for pprecursor, peptide and proteins.

It's a breaking chnage because we start introducing semantic column names like:
`pg => pg.intensity, precursor.intensity, peptide.intensity`
@mschwoer 
It's also a breaking change because we drop `peptide_level_lfq` and `precursor_level_lfq` as it will always be performed.

### CODE REVIEW

File: alphadia/outputtransform/search_plan_output.py [hunk 1]
```diff
@@ -32,6 +32,13 @@
 logger = logging.getLogger()
```
Comment (mschwoer) [inline alphadia/outputtransform/search_plan_output.py:32]: In which way is this change breaking? because it adds new columns to the output tables?

File: alphadia/outputtransform/search_plan_output.py [hunk 2]
```diff
@@ -618,6 +608,51 @@ class LFQOutputConfig:
...
+            Updated precursor table with merged quantification data
+        """
+        intensity_column_mapping = {
+            "precursor": "precursor_intensity",
+            "peptide": "peptide_intensity",
+            "pg": "intensity",
+        }
+
+        for config in quantlevel_configs:
+            lfq_df = lfq_results.get(config.level_name, pd.DataFrame())
```
Comment (mschwoer) [inline alphadia/outputtransform/search_plan_output.py:640]:
```
            lfq_df = lfq_results.get(config.level_name)

            if lfq_df is None:
                continue
```

File: alphadia/outputtransform/search_plan_output.py [hunk 3]
```diff
@@ -618,6 +608,51 @@ class LFQOutputConfig:
...
+
+            if len(lfq_df) == 0:
+                continue
+
+            intensity_column = intensity_column_mapping.get(
+                config.level_name, f"{config.level_name}_intensity"
```
Comment (mschwoer) [inline alphadia/outputtransform/search_plan_output.py:646]: is the default required? you have e.g. `"precursor": "precursor_intensity",` explicitly defined

File: tests/unit_tests/outputtransform/test_search_plan_output.py [hunk 4]
```diff
@@ -184,3 +184,115 @@ def test_output_transform():
...
 
     shutil.rmtree(temp_folder)
+
+
+def test_merge_quant_levels_to_psm_merges_precursor_level():
+    """Test that precursor level quantification is merged correctly."""
+    from alphadia.outputtransform.search_plan_output import (
+        LFQOutputConfig,
+        SearchPlanOutput,
+    )
```
Comment (mschwoer) [inline tests/unit_tests/outputtransform/test_search_plan_output.py:194]: import to top please

File: tests/unit_tests/outputtransform/test_search_plan_output.py [hunk 5]
```diff
@@ -184,3 +184,115 @@ def test_output_transform():
...
+
+    spo = SearchPlanOutput({"general": {"save_figures": False}}, "/tmp")
```
Comment (mschwoer) [inline tests/unit_tests/outputtransform/test_search_plan_output.py:196]: `search_plan_output` .. less mental mapping >> less characters

Comment (mschwoer) [inline tests/unit_tests/outputtransform/test_search_plan_output.py:196]: `/tmp` => `/some_output_path`

File: tests/unit_tests/outputtransform/test_search_plan_output.py [hunk 6]
```diff
@@ -184,3 +184,115 @@ def test_output_transform():
...
+    psm_df = pd.DataFrame({"mod_seq_charge_hash": ["A1"], "run": ["run1"]})
+    lfq_results = {
+        "precursor": pd.DataFrame({"mod_seq_charge_hash": ["A1"], "run1": [100.0]})
+    }
+    configs = [LFQOutputConfig("mod_seq_charge_hash", "precursor")]
+
+    result = spo._merge_quant_levels_to_psm(psm_df, lfq_results, configs)
+
+    assert "precursor_intensity" in result.columns
```
Comment (mschwoer) [inline tests/unit_tests/outputtransform/test_search_plan_output.py:205]: this assert is redundant

File: tests/unit_tests/outputtransform/test_search_plan_output.py [hunk 7]
```diff
@@ -184,3 +184,115 @@ def test_output_transform():
...
+    assert result["precursor_intensity"].iloc[0] == 100.0
```
Comment (mschwoer) [inline tests/unit_tests/outputtransform/test_search_plan_output.py:206]: could be made stronger:
```
assert result["precursor_intensity"].values == [100.0]
```

File: tests/unit_tests/outputtransform/test_search_plan_output.py [hunk 8]
```diff
@@ -184,3 +184,115 @@ def test_output_transform():
...
+
+    spo = SearchPlanOutput({"general": {"save_figures": False}}, "/tmp")
+    psm_df = pd.DataFrame({"pg": ["PG1"], "run": ["run1"]})
+    lfq_results = {"pg": pd.DataFrame({"pg": ["PG1"], "run1": [700.0]})}
+    configs = [LFQOutputConfig("pg", "pg")]
+
+    result = spo._merge_quant_levels_to_psm(psm_df, lfq_results, configs)
+
+    assert "intensity" in result.columns
+    assert result["intensity"].iloc[0] == 700.0
```
Comment (mschwoer) [inline tests/unit_tests/outputtransform/test_search_plan_output.py:242]: are these tests really adding something? if it works for one level, it should work for all..
I would rather add a test with multiple runs, or some edge cases
